examples/allennlp/classifier.jsonnet

```diff
@@ -68,7 +67,7 @@ local ENCODER = CNN_FIELDS(
     },
     patience: 2,
     validation_metric: '+accuracy',
-    epoch_callbacks: [
+    callbacks: [
```

Comment:
I think this example only works with `allennlp>=2.0.0` since `allennlp==1.5.0` does not have the `callbacks` argument ([callback-related arguments](https://github.com/allenai/allennlp/blob/e09cf211d1be7aa6eabd3ff641bf2cac6ea1c5dc/allennlp/training/trainer.py#L425-L438)). If so, how about raising the error [here](https://github.com/optuna/optuna/blob/0a59ab0629288354f5f70c0ea41614b600523c53/examples/allennlp/allennlp_jsonnet.py#L45-L46)?

Reply:
It actually raised the following error with `allennlp==1.5.0`:

```console
allennlp.common.checks.ConfigurationError: Extra parameters passed to GradientDescentTrainer: {'callbacks': [{'type': 'optuna_pruner'}]}
```

Reply:
You're right, thank you for pointing it out. We can run this example with AllenNLP v1.x.x if `callback` is removed from `classifier.jsonnet`. So choices are (a) requiring v2.x.x with the current example or (b) requiring >= 1.0.0 with removing callback from config. I'm wondering which is better and I'd like your opinion.

Reply:
Thank you for your reply.
I think the callbacks in `jsonnet` is one of the key features of this integration, and I prefer (a). Since AllenNLP v1.x.x users need to use Optuna v2.5.0, how about showing the message, for example, as follows:

```
If you want to use `allennlp<2.0.0`, please install `optuna==2.5.0` and refer to the following example:
https://github.com/optuna/optuna/blob/v2.5.0/examples/allennlp/allennlp_jsonnet.py
```

Reply:
Thank you for the suggestion. I added version checks to examples in 53e5c33.